osdk/src/base_crate/mod.rs

```diff
@@ -124,9 +141,38 @@ fn copy_profile_configurations(workspace_root: impl AsRef<Path>) {
     // Copy the profile configurations
     let profile = target_manifest.get("profile");
     if let Some(profile) = profile {
-        manifest.insert("profile".to_string(), profile.clone());
+        manifest.insert(
+            "profile".to_string(),
+            toml::Value::Table(profile.as_table().unwrap().clone()),
+        );
+    }
+
+    let content = toml::to_string(&manifest).unwrap();
+    fs::write(manifest_path, content).unwrap();
+}
+
+fn add_feature_entries(dep_crate_name: &str, features: &toml::Table) {
+    let manifest_path = "Cargo.toml";
+    let mut manifest: toml::Table = {
+        let content = fs::read_to_string(manifest_path).unwrap();
+        toml::from_str(&content).unwrap()
+    };
+
+    let mut table = toml::Table::new();
+    for (feature, value) in features.iter() {
+        let value = if feature != &"default_features".to_string() {
```

Comment:
Um, you remind me that we still need to keep "default" entry and add "no-default-features" to the dependency spec since the user may switch off default features.

Reply:
I agree keeping `dafault` entry to support switching off default features.

But the key should be `default`, instead of `default_features`, right?. From the example of [Cargo book](https://doc.rust-lang.org/cargo/reference/features.html):
```toml
[features]
default = ["ico", "webp"]
bmp = []
png = []
ico = ["bmp", "png"]
webp = []
```

Further, is `no-default-features` added somewhere? I didn't find it.

Reply:
Emm I remember that I fixed it
Maybe memory blurred

Reply:
Seems forgot to stash pop. I will take a review myself after fixing it.